File: python-wrapper/src/python/crypto/entities/ShareMetadataBehaviour.py
from typing import Union, Dict
from enum import Enum

class ShareMetadataBehaviour(Enum):
	Required = "REQUIRED"
	IfAvailable = "IF_AVAILABLE"
	Never = "NEVER"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'ShareMetadataBehaviour':
		if data == "REQUIRED":
			return ShareMetadataBehaviour.Required
		elif data == "IF_AVAILABLE":
			return ShareMetadataBehaviour.IfAvailable
		elif data == "NEVER":
			return ShareMetadataBehaviour.Never
		else:
			raise Exception(f"{data} is not a valid value for ShareMetadataBehaviour enum.")

__all__ = ['ShareMetadataBehaviour']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/crypto/entities/ShareMetadataBehaviour.kt
package com.icure.cardinal.sdk.crypto.entities

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Specifies a behaviour for the sharing of encryption keys or owning entity ids in the extended apis 'share' methods.
 */
@Serializable
enum class ShareMetadataBehaviour {
	/**
	 * The method must share the metadata with the delegate. If this is not possible, because for example the current user has no access to this kind of
	 * metadata, the method will throw an error.
	 */
	@SerialName("REQUIRED")
	Required,
	/**
	 * The method must share the metadata with the delegate if available. If this is not possible, because for example the current user has no access to
	 * this kind of metadata, the method will simply not share this metadata.
	 */
	@SerialName("IF_AVAILABLE")
	IfAvailable,
	/**
	 * The method must not share the metadata with the delegate, even if the current data owner can access it.
	 */
	@SerialName("NEVER")
	Never
}
